### DISCUSSION TITLE
Available timeframe is shorter than the specified number of hours to operate. Optimization will fail.

### DISCUSSION THREAD

**Discussion by purcell-lab** [davidusb-geek/emhass#384] [General]
@michaelpiron ,  I am seeing an error when the available timeframe is shorter than the end time step. Resulting in an infeasible optimisation.

My recollection is this used to soft fail with the validated optimisation window, but this is no longer the case:
https://emhass.readthedocs.io/en/latest/lpems.html#time-windows-for-deferrable-loads

```logs
2024-11-23 15:15:39,922 - web_server - WARNING - Deferrable load 5 : Available timeframe is shorter than the specified number of hours to operate. Optimization will fail.
2024-11-23 15:15:39,923 - web_server - DEBUG - Deferrable load 5: Validated optimization window: 0 --> 3
2024-11-23 15:15:40,671 - web_server - INFO - Status: Infeasible
```

**Comment by michaelpiron** [comment]
Hi Mark, not sure what your question is exactly.

At the time, I introduced in the codebase the feature to define start & end timeframes for deferrable loads.
With that, I detected that there is a possible "unhappy flow" usecase scenario where the defined timeframe for a deferrable load is shorter than the required number of hours for the deferrable load to operate. In that scenario, we know for certain that the optimization will fail. Therefore, I foresaw a warning log message to inform the user, so they would understand the reason why the optimization fails.

As I stated, I'm not sure to have understood your question. Could you elaborate a bit more?
Thanks.
